test/Microsoft.Azure.Devices.Edge.Test.Common/Linux/EdgeDaemon.cs

```diff
@@ -0,0 +1,209 @@
+                async () =>
+                {
+                    await this.InternalStopAsync(token);
+
+                    const string YamlPath = "/etc/iotedge/config.yaml";
+                    string text = await File.ReadAllTextAsync(YamlPath, token);
+
+                    var doc = new YamlDocument(text);
+                    doc.ReplaceOrAdd("provisioning.device_connection_string", deviceConnectionString);
+                    doc.ReplaceOrAdd("hostname", hostname);
+
+                    string result = doc.ToString();
+
+                    FileAttributes attr = 0;
+                    if (File.Exists(YamlPath))
+                    {
+                        attr = File.GetAttributes(YamlPath);
+                        File.SetAttributes(YamlPath, attr & ~FileAttributes.ReadOnly);
+                    }
+
+                    await File.WriteAllTextAsync(YamlPath, result, token);
+
+                    if (attr != 0)
+                    {
+                        File.SetAttributes(YamlPath, attr);
+                    }
+
+                    await this.InternalStartAsync(token);
+                },
+                "Configured edge daemon for device '{Device}' registered as '{Id}'",
+                hostname,
+                builder.DeviceId);
+        }
+
+        public Task StartAsync(CancellationToken token)
+        {
+            return Profiler.Run(
+                () => this.InternalStartAsync(token),
+                "Started edge daemon");
+        }
+
+        async Task InternalStartAsync(CancellationToken token)
+        {
+            string[] output = await Process.RunAsync("systemctl", "start iotedge", token);
+            Log.Verbose(string.Join("\n", output));
+            await this.WaitForStatusAsync(ServiceControllerStatus.Running, token);
+        }
+
+        public Task StopAsync(CancellationToken token)
+        {
+            return Profiler.Run(
+                () => this.InternalStopAsync(token),
+                "Stopped edge daemon");
+        }
+
+        async Task InternalStopAsync(CancellationToken token)
+        {
+            string[] output = await Process.RunAsync("systemctl", "stop iotedge", token);
+            Log.Verbose(string.Join("\n", output));
+            await this.WaitForStatusAsync(ServiceControllerStatus.Stopped, token);
+        }
+
+        public async Task UninstallAsync(CancellationToken token)
+        {
+            try
+            {
+                await this.InternalStopAsync(token);
+            }
+            catch (Win32Exception e)
+            {
+                Log.Verbose(e, "Failed to stop edge daemon, probably because it is already stopped");
+            }
+
+            try
+            {
+                await Profiler.Run(
+                    async () =>
+                    {
+                        string[] output =
+                            await Process.RunAsync("apt-get", "purge --yes libiothsm-std", token);
```

Comment:
we call apt-get purge for libiothsm, because iotedge daemon depends on libiothsm; therefore it will uninstall iotedge daemon as well, do I understand correctly?

Reply:
That's correct